snapcraft/tests/fixture_setup.py

```diff
@@ -15,9 +15,13 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import os
+import threading
```

Comment:
Here be dragons.

Trying to run a server and a client inside the same process triggers all kind of racy and hanging behaviours.
Been there 10 years ago (literally), debugged dozen of bugs, in a test suite involving thousands of such servers running in threads, and I'm still seeing occasional issues.

If you want fake servers, run them in a separate process once you've debugged them thoroughly since once they are in a separate process they become way harder to debug.